src/components/network/NetworkList.vue

```diff
@@ -32,9 +32,21 @@ export default {
   },
   methods: {
     selectNetworkHandler(network) {
-      if (this.connection.network !== network.id) {
+      let confirm = this.confirmModalOpen()
+      if (this.connection.network !== network.id && confirm) {
         this.$store.dispatch(`setNetwork`, network)
       }
+    },
+    confirmModalOpen() {
+      let confirm = false
+      if (localStorage.getItem(`session`)) {
```

Comment:
how about checking the vuex getter `address` or `session.signedIn`? Check `DelegationModal.vue` for example